Controllers/HomeController.cs

```diff
@@ -3,21 +3,88 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
+using System.Text.RegularExpressions;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Newtonsoft.Json;
 
 namespace Zoltu.Bags.Client.Controllers
 {
 	public class HomeController : Controller
 	{
+		private static Regex productRegx = new Regex(@"product\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex tagRegx = new Regex(@"tags\/(.*?)\/*?$", RegexOptions.IgnoreCase);
+		private static Regex minPriceRegx = new Regex(@"minprice\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex maxPriceRegx = new Regex(@"maxprice\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex aboutRegx = new Regex(@"aboutus", RegexOptions.IgnoreCase);
+		private static HttpClient client = null;
+
+		public HomeController()
+		{
+			// Best practices: https://www.asp.net/web-api/overview/advanced/calling-a-web-api-from-a-net-client
+			client = new HttpClient();
```

Comment:
> we want to dispose http client after every call

I don't believe this is the case based on the link you provided in the comment above: https://www.asp.net/web-api/overview/advanced/calling-a-web-api-from-a-net-client

> HttpClient is intended to be instantiated once and re-used throughout the life of an application

What I am recommending is this:
```
public class HomeController : Controller
{
	private static readonly HttpClient httpClient = new HttpClient();
	static HomeController
	{
		client.BaseAddress = new Uri("https://bags-api.zoltu.com/api/");
		client.DefaultRequestHeaders.Accept.Clear();
		client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
	}
}
```
This example would result in one `HttpClient` for the entire application.  If you initialize the `HttpClient` in the normal constructor (not a static one), then you end up with one `HttpClient` per request which is what the article you linked recommends *against*.  For the same reason, the `HttpClient` should not be disposed at the end of the request, it should live for the life of the application.